src/components/ArticleList.js

```diff
@@ -30,6 +48,34 @@ const ArticleList = ({ topic }) => {
     [ topic, isLoading ]
   );
 
+  const handleChange = (e) => {
+    if (!topic) {
+      getArticlesSortNoTopic(e.target.value).then((sortedArticles) => {
+        setArticles(sortedArticles.articles);
+        setSort(e.target.value);
+      });
+    } else {
+      getArticlesSort(topic, e.target.value).then((sortedArticles) => {
+        setArticles(sortedArticles.articles);
+        setSort(e.target.value);
+      });
+    }
+  };
+
```

Comment:
If we change the sort, then we make an immediate request using the sort value but we don't use the current value of order... likewise the other way round. Since you are holding both of these things in state, you do have access to their current value.

Reply:
Could we just trigger the useEffect if sort or order changes?